### PR TITLE
Include nulls in dictionary block logical size

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java [hunk 1]
```diff
@@ -271,7 +271,7 @@ public long getPositionsSizeInBytes(boolean[] positions)
                 used[getId(i)] = true;
             }
         }
-        return dictionary.getPositionsSizeInBytes(used) + Integer.BYTES * countUsedPositions(positions);
+        return dictionary.getPositionsSizeInBytes(used) + (Integer.BYTES * (long) countUsedPositions(positions));
```
Comment (findepi) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java]: the added parens are redundant

File: presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java [hunk 2]
```diff
@@ -523,4 +523,10 @@ protected static void testIncompactBlock(Block block)
         assertNotCompact(block);
         testCopyRegionCompactness(block);
     }
+
+    @SuppressWarnings("unchecked")
+    private static <T> T[] newArrayOfType(T[] array, int length)
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java]: I would directly call `java.util.Arrays#copyOf(T[], int)`. Optionally followed by `Arrays.fill(.., null)` (which is not strictly necessary, cause you set all the elements anyway)

File: presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java (hunk 1 above)
Comment (electrum) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java]: They are redundant, in that they don't change the behavior, but I find it improves readability and serves to clarify the intention.

File: presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java (hunk 2 above)
Comment (electrum) [reply presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java]: Good idea, I'll update to that

**Pull request merged**